pkg/channel/consolidated/dispatcher/dispatcher.go

```diff
@@ -241,6 +241,7 @@ func (d *KafkaDispatcher) UpdateHostToChannelMap(config *Config) error {
 		return errors.New("nil config")
 	}
 
+	// TODO why we have a lock inside another lock for this map?!
```

Comment:
have we gh issue for track this, as an investigtation?

Reply:
Nope, should I create one?